components/html-block/html-block.js

```diff
@@ -178,24 +175,31 @@ class HtmlBlock extends LitElement {
 		super.connectedCallback();
 		if (this._contextObserverController) this._contextObserverController.hostConnected();
 
-		if (!this._templateObserver || this.noDeferredRendering) return;
+		if (!this._contentObserver || this.noDeferredRendering) return;
 
 		const template = this._findSlottedElement('TEMPLATE');
-		if (template) this._templateObserver.observe(template.content, { attributes: true, childList: true, subtree: true });
+		if (template) this._contentObserver.observe(template.content, { attributes: true, childList: true, subtree: true });
+		else {
+			const slot = this.shadowRoot.querySelector('slot');
+			if (slot) {
+				slot.assignedNodes({ flatten: true }).forEach(
+					node => this._contentObserver.observe(node, { attributes: true, childList: true, subtree: true })
+				);
+			}
+		}
 	}
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
 		if (this._contextObserverController) this._contextObserverController.hostDisconnected();
-		if (this._templateObserver) this._templateObserver.disconnect();
+		if (this._contentObserver) this._contentObserver.disconnect();
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
 		if (this._renderContainer) return;
-
-		this.shadowRoot.innerHTML += `<div class="d2l-html-block-rendered${this.compact ? ' d2l-html-block-compact' : ''}"></div><slot></slot>`;
+		this.shadowRoot.innerHTML += `<div class="d2l-html-block-rendered${this.compact ? ' d2l-html-block-compact' : ''}"></div><slot ${!this.noDeferredRendering ? 'style="display: none"' : ''}></slot>`;
```

Comment:
The [classMap](https://lit.dev/docs/v1/lit-html/template-reference/#classmap) and [styleMap](https://lit.dev/docs/v1/lit-html/template-reference/#stylemap) might be a slightly cleaner way to do the dynamic CSS classes and inline styles.

Reply:
Hmm, I don't think that works outside of a Lit template though, does it? I did try that, but all it did was spit out the actual mapping code into the html-block.

Reply:
Ah right, sorry.

Reply:
No worries, I wasn't 100% sure I wasn't missing something. I'm also hopeful that once all of this is done and dusted, I can take a look at switching this over to use Lit's rendering and template this stuff. I'm cautiously optimistic that it may be possible with more selective math rendering, but... No promises. 😅